src/python/dxpy/__init__.py

```diff
@@ -188,6 +191,35 @@ def _process_method_url_headers(method, url, headers):
     else:
         return method, _url, _headers
 
+
+_RETRYABLE_SOCKET_ERRORS = set([errno.ENETDOWN, errno.ENETUNREACH, errno.ECONNREFUSED])
+
+
+def _is_retryable_exception(e):
+    """Returns True if the exception is always safe to retry.
+
+    This is True if the client was never able to establish a connection
+    to the server (for example, name resolution failed or the connection
+    could otherwise not be initialized).
+
+    Conservatively, if we can't tell whether a network connection could
+    have been established, we return False.
+
+    """
+    try:
+        if isinstance(e, requests.exceptions.ConnectionError):
+            # Unfortunately requests doesn't seem to provide a sensible
+            # API to retrieve the cause
+            cause = e.args[0].args[1]
```

Comment:
Indeed. But even without the special syntax, is there no Python convention to retrieve the original exception? I guess what I had in mind was like this in Java: http://docs.oracle.com/javase/7/docs/api/java/lang/Throwable.html#getCause()